src/EVM/Expr.hs

```diff
@@ -1543,25 +1543,19 @@ constFoldProp ps = oneRun ps (ConstState mempty True)
   where
     oneRun ps2 startState = (execState (mapM (go . simplifyProp) ps2) startState).canBeSat
     go :: Prop -> State ConstState ()
-    go x = case x of
+    go = \case
         -- PEq
         PEq (Lit l) a -> do
           s <- get
           case Map.lookup a s.values of
-            Just l2 -> case l==l2 of
-                True -> pure ()
-                False -> put ConstState {canBeSat=False, values=mempty}
-            Nothing -> do
-              let vs' = Map.insert a l s.values
-              put $ s{values=vs'}
+            Just l2 -> unless (l==l2) $ put ConstState {canBeSat=False, values=mempty}
+            Nothing -> modify (\s' -> s'{values=Map.insert a l s'.values})
```

Comment:
This should be OK, no? We do *not* need to extract `Map.insert a l s'.values` into a variable first, no?

Reply:
Sure this is good :)